packages/wdio-appium-service/src/launcher.ts

```diff
@@ -199,13 +199,40 @@ export default class AppiumLauncher implements Services.ServiceInstance {
              * Ensure all child processes are also killed
              */
             log.info('Killing entire Appium tree')
-            treeKill(this._process.pid, 'SIGTERM', (err) => {
-                if (err) {
-                    return log.warn('Failed to kill process:', err)
-                }
-
+            try {
+                // First attempt with SIGTERM
+                await new Promise<void>((resolve, reject) => {
+                    treeKill(this._process!.pid!, 'SIGTERM', (err) => {
+                        if (err) {
+                            reject(err)
+                            return
+                        }
+                        resolve()
+                    })
```

Comment:
Can we move

```ts
(err) => {
                        if (err) {
                            reject(err)
                            return
                        }
                        resolve()
                    }
```
into a reusable utility function, as we use it twice here?

Reply:
@christian-bromann Thanks for quick review. Addressed the issue.

Please check